Assets/Scripts/Effects/AnimateWater.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Effects
+{
+    public class AnimateWater : MonoBehaviour
+    {
+        [SerializeField] [Tooltip("Determines the difference between the peak and bottom of a wave")]
+        private float power = 3f;
+    
+        [SerializeField] [Tooltip("Determines the size of waves")]
+        private float scale = 1f;
+    
+        [SerializeField] [Tooltip("Determines how fast waves aree")]
+        private float frequency = 1f;
+
+        private float _xOffset, _yOffset;
+        private MeshFilter _mf;
+
+
+        void Start()
+        {
+            _mf = GetComponent<MeshFilter>();
+            AnimateWaves();
+        }
+    
+        void Update()
+        {
+            AnimateWaves();
+            _xOffset += Time.deltaTime * frequency;
+            if(_yOffset <= 0.3) _yOffset += Time.deltaTime * frequency;
+            if (_yOffset >= power) _yOffset -= Time.deltaTime * frequency;
+        }
+    
+        private void AnimateWaves()
+        {
+            Vector3[] vertices = _mf.mesh.vertices;
+
+            for (int i = 0; i < vertices.Length; i++)
+            {
+                vertices[i].y = CalculateHeight(vertices[i].x, vertices[i].z) * power;
+            }
+
+            _mf.mesh.vertices = vertices;
+        }
+
+        private float CalculateHeight(float x, float y)
+        {
+            float xCord = x * scale + _xOffset;
+            float yCord = y * scale + _yOffset;
```

Comment:
Why do you have two variables for the size of the wave scale and power? Is it necessary?

Reply:
Power describes the "height" of the wave, as in the peak-to-peak amplitude of a sine wave.
Scale is the size of the wave as in the frequency of a sine wave.

Reply:
> the size of the wave as in the frequency of a sine wave
I don't really get it, scale is the wavelength? But if you have a reason....